lib/hotel.rb

```diff
@@ -0,0 +1,183 @@
+require_relative "room"
+require_relative "reservation"
+require_relative "date_range"
+require_relative "hotel_block"
+require "date"
+
+module HotelSystem
+  class Hotel
+    attr_reader :rooms, :blocks
+    attr_accessor :reservations
+
+    def initialize(number_of_rooms)
+      @rooms = []
+      add_rooms(number_of_rooms)
+      @reservations = []
+      @blocks = []
+    end
+
+    def add_rooms(number)
+      if number.class != Integer || number < 1
+        raise ArgumentError, "Please enter a number greater than 1."
+      end
+      (1..number).each do |num|
+        room = HotelSystem::Room.new(num)
+        @rooms << room
+      end
+    end
+
+    def valid_date_entry?(year, month, day)
+      HotelSystem::DateRange.valid_date_entry?(year, month, day)
+    end
+
+    def create_date_range(start_year:, start_month:, start_day:, num_nights: nil)
+      valid_date_entry?(start_year, start_month, start_day)
+      dates = HotelSystem::DateRange.new(start_year: start_year, start_month: start_month, start_day: start_day, num_nights: num_nights)
+      return dates
```

Comment:
I like the idea of wrapping this in a method, good work practicing what we see in Metz.

However, in this case I don't know that it gains you much. This method has the same signature as `DateRange.new`, so you're not reducing complexity, and since the constructor for `DateRange` calls `valid_date_entry?` already the check here is redundant.